frame/staking/src/lib.rs

```diff
@@ -2201,11 +2201,14 @@ decl_module! {
 				score,
 				era,
 				size,
-			)
-			// TODO: instead of returning an error, panic. This makes the entire produced block
-			// invalid.
-			// This ensures that block authors will not ever try and submit a solution which is not
-			// an improvement, since they will lose their authoring points/rewards.
+			).expect(
+				"An unsigned solution can only be submitted by validators; A validator should \
+				always produce correct solutions, else this block should not be imported, thus \
+				effectively depriving the validators from their authoring reward. Hence, this panic
+				is expected."
+			);
+
+			Ok(None.into())
```

Comment:
Well maybe we should refund here? but does it really matter? if this block is being baked, it probably already took a lot of time to build the solution and now we realise that it was wrong, so we should consume weight.

Reply:
you my comment was BS :D

Reply:
Right I forgot about the adjustments that we return from the inner function.